cl/opinion_page/views.py

```diff
@@ -1119,12 +1120,21 @@ async def citation_redirector(
     if not volume and not page:
         citations = eyecite.get_citations(reporter)
         if citations and citations[0].groups:
-            c = citations[0]
+            # check whether the parsed object is an opinion citation
+            if not isinstance(
+                citations[0], (FullCaseCitation, ShortCaseCitation)
+            ):
+                return await throw_404(
+                    request,
+                    {"not_opinion_citation": True, "private": False},
+                )
+
```

Comment:
Can't we do better here? Like, if the first citation found is not the correct type, grab the next until we find a good one? 

This input should work:

```
§102 USC 222 is the statute that was discussed in 22 U.S. 444
```

The statute should be found, ignored, and then then the case citation should be used.

Reply:
Thank you for pointing this out. I'll implement the same approach we use for the API. 👍

Reply:
Here's a gif showing how the filtering logic works:

![filter out non-caselaw](https://github.com/freelawproject/courtlistener/assets/55959657/f8940d7e-28d8-4200-a2d2-7c408a05cd28)